### PR TITLE
divisors: add doctest filter

### PR DESCRIPTION
**Pull request by fingolfin**
... to filter out annotations added to the doctest output

Let's see how that works, I may have to tweak it.

### CODE REVIEW

File: src/Primes.jl [hunk 1]
```diff
@@ -968,7 +968,7 @@ For convenience, `divisors(0)` returns `[]`.
 
 # Example
 
-```jldoctest
+```jldoctest; filter = r"\s+#.*$"
```
Comment (fingolfin) [inline src/Primes.jl:971]:
```suggestion
```jldoctest; filter = r"\s+#.*\$"
```
Ugh. No need for quoting in a `r` string. However, this is actually occurring in a docstring, so we *do* need to quote... *sigh*

File: src/Primes.jl [hunk 2]
```diff
@@ -968,7 +968,7 @@ For convenience, `divisors(0)` returns `[]`.
 
 # Example
 
-```jldoctest
+```jldoctest; filter = r"\s+#.*\$"
```
Comment (fingolfin) [inline src/Primes.jl:971]:
```suggestion
```jldoctest; filter = r"\\s+#.*\$"
```

File: src/Primes.jl [hunk 3]
```diff
@@ -968,7 +968,7 @@ For convenience, `divisors(0)` returns `[]`.
 
 # Example
 
-```jldoctest
+```jldoctest; filter = r"\\s+#.*\$"
```
Comment (fingolfin) [inline src/Primes.jl:971]:
```suggestion
```jldoctest; filter = r"\\s+#.*"
```

**Comment by fingolfin** [comment]
It seems I have hit a Documenter bug or two. I'll make an alternate PR for now until I can get this here properly resolved.

File: src/Primes.jl [hunk 4]
```diff
@@ -968,7 +968,7 @@ For convenience, `divisors(0)` returns `[]`.
 
 # Example
 
-```jldoctest
+```jldoctest; filter = r"\\s+#.*"
```
Comment (fingolfin) [inline src/Primes.jl:971]:
```suggestion
```jldoctest; filter = r"(\\s+#.*)?"
```

**Comment by fingolfin** [comment]
Got this working now, @oscardssmith

**Comment by oscardssmith** [comment]
Many thanks!

**Pull request merged**